src/pages/Doeun/Login/Components/LoginBox/LoginBox.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import Main from '../../../Main/Main';
+import './LoginBox.scss';
+
+class LoginBox extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      id: '',
+      pw: '',
+      type: 'password',
+      warning: '',
+      validId: /\W/,
+    };
+  }
+
+  handleInputId = e => {
+    this.setState({ id: e.target.value });
+  };
+
+  handleInputPw = e => {
+    this.setState({ pw: e.target.value });
+  };
+
+  goToMain = () => {
+    this.props.history.push('/main-doeun');
+  };
+
+  showPassword = () => {
+    this.state.type === 'password'
+      ? this.setState({ type: 'text' })
+      : this.setState({ type: 'password' });
+  };
```

Comment:
- 국! 👍
- 다만, **변하는 부분에 대해서만** 삼항 연산자를 적용해주시면 더 좋을 것 같습니다.
```suggestion
  showPassword = () => {
    const { type } = this.state;
    this.setState({
      type: type === 'password' ? 'text' : 'password'
    })
  };
```

Reply:
앗 오브젝트 안에 바로 저렇게 쓸 수 있군요!! 감사합니다~~~~~